app/views.py

```diff
@@ -29,19 +29,19 @@ def generate_forecast():
     if not place:
         return 'No address provided.', 400
 
-    geocode = Geocode(address=place)
-    is_place_valid = geocode.is_place_query_valid()
-
-    if not is_place_valid:
-        return 'No results found. Try a more generic place name i.e. local, province', 400
-
     frequency = 'currently'  # default forecast set to current
-    valid_frequency = ['currently', 'hourly']
+    valid_frequency = ['currently', 'hourly', 'daily']
     last_query = place.split(' ')[-1].lower()
     if last_query in valid_frequency:
         place = place.replace(last_query, '')  # replace time method(if present) with empty string so that only the place remains. Example query: cebu hourly
         frequency = last_query
 
+    geocode = Geocode(address=place)
+    is_place_valid = geocode.is_place_query_valid()
```

Comment:
I only switched their position because I realized that it determines if the place is valid even without slicing out the frequency. What I did here is just to make sure that is slices first the frequency before validating the address.

Ex: Cebu, City daily
                           ^
                         needs to extract this first then validate the address